echopype/convert/parse_base.py

```diff
@@ -495,7 +495,12 @@ def _read_datagrams(self, fid):
 
             # XML datagrams store environment or instrument parameters for EK80
             if new_datagram["type"].startswith("XML"):
-                if new_datagram["subtype"] == "environment":
+                # Check that environment datagrams aren't invalid
+                # Temporary fix for suspected EK80 software bug described in:
+                # https://github.com/OSOceanAcoustics/echopype/issues/1386
+                if new_datagram["subtype"] == "environment" and set(
+                    ["drop_keel_offset", "drop_keel_offset_is_manual"]
+                ) != set(new_datagram["environment"].keys()):
```

Comment:
So I don't think this is a "bug" per se - if the EK80 software only uses variables existing in the data to update computation, that is entirely valid. It is us that need to handle the variability.

The part that I referred to as a potential bug is that the replayed environment datagram has the same timestamp of the original time the data was collected -- but this may actually be intentional, since in a replayed scenario there is no environmental data being collected.
```suggestion
                # Check that environment datagrams contain more than
                # just drop_keel_offset and drop_keel_offset_is_manual
                # Temporary fix for handling >1 EK80 environment datagrams described in:
                # https://github.com/OSOceanAcoustics/echopype/issues/1386
                if new_datagram["subtype"] == "environment" and set(
                    ["drop_keel_offset", "drop_keel_offset_is_manual"]
                ) != set(new_datagram["environment"].keys()):
```